Assignments/HW2/hw2.md

```diff
@@ -0,0 +1,166 @@
+
+Avery has already decided on the API's specification, and in fact has already also created a client that coforms to that specification - so you will be able to start testing your implementation right away. Here is the specification of the new route that you need to add for this API:
+* URL: `/towns/:townID/conversationAreas` (where `:townID` is a parameter that specified the town in which the conversation area should be created)
+* HTTP Method: `post`
+* Body: JSON, the body matches the type `ConversationAreaCreateRequest`, defined in `src/client/TownsServiceClient.ts` and reproduced below
+
+```ts
+/**
+ * Payload sent by the client to create a new conversation area
+ */
+export interface ConversationAreaCreateRequest {
+  coveyTownID: string;
+  sessionToken: string;
+  conversationArea: ServerConversationArea;
+}
+```
+
+Avery's sanity test for this task uses their API client to make a request to create a conversation area. You can run it right now, by running the command `npm test CoveyTownConversationAPI`. Before you implement this task, you should expect to see the tests fail, with a message `Request failed with status code 404` (404 is the error code that indicates that an address does not exist). 
+
+**The route**. Following the example routes that already exist in `router/towns.ts`, create the new route as specified above in the file `router/towns.ts`. Your route should follow the conventions in the other routes: forward the request on to the `conversationAreaCreateHandler` and `await` the response. If `conversationAreaCreateHandler` returns `true`, return the response with the HTTP status code `OK`. If `conversationAreaCreateHandler` throws an error, log the error and return a response with HTTP status `INTERNAL_SERVER_ERROR`, and the message `Internal server error, please see log in server for more details`.
+
+The router interfaces with the HTTP server through the variable `app`, which in our case, is a library called [ExpressJS](https://expressjs.com). For tips on working with Express, see the [ExpressJS routing docs](https://expressjs.com/en/guide/routing.html)
+
+📝 Check your work: When you run the sanity test, it should still fail, but this time with the message "Error processing request: This feature is not yet implemented" (the default behavior of `conversationAreaCreateHandler`). If you get this message, then you have successfully completed thisf irst task!
+
+### Task 1.2: Implement business logic to create a conversation area
+With the top of the architecture stack implemented, it's now time to implement some business logic to actually create a conversation area when the `conversationAreaCreateHandler` is called. This handler should:
+1. Validate that the supplied session token is a valid token for the specified town
+2. Delegate the actual responsibility for creating the town to the `CoveyTownController`, invoking the `addConversationArea` method on the appropriate town controller.
```

Comment:
My assumption was that students should be adding conversation areas to existing towns, so the sentence _"actual responsibility for creating the **town** to the"_ does not fit right in. Maybe it should say something like _"actual responsibility for creating the **conversation area** to the"_?

Reply:
Thanks! This was a typo, fixed n ow